src/MessageCreationUtil.js

```diff
@@ -114,10 +120,21 @@ export default class MessageCreationUtil {
         const sequenceNumber = this.getNextSequenceNumber(key, timestamp)
         const streamMessage = StreamMessage.create(
             [stream.id, streamPartition, timestamp, sequenceNumber, publisherId, this.msgChainId], this.getPrevMsgRef(key),
-            StreamMessage.CONTENT_TYPES.JSON, data, StreamMessage.SIGNATURE_TYPES.NONE, null,
+            StreamMessage.CONTENT_TYPES.MESSAGE, StreamMessage.ENCRYPTION_TYPES.NONE, data, StreamMessage.SIGNATURE_TYPES.NONE, null,
         )
         this.publishedStreams[key].prevTimestamp = timestamp
         this.publishedStreams[key].prevSequenceNumber = sequenceNumber
+
+        if (groupKey && this.groupKeys[stream.id]) {
```

Comment:
Should also check that the key really is new? `groupKey !== this.groupKeys[stream.id]`

(may also affect the `else` branch below)